src/transforms/babel/babelrc.js

```diff
@@ -0,0 +1,83 @@
+const {buildRootChain} = require('@babel/core/lib/config/config-chain');
+const {mergeOptions} = require('@babel/core/lib/config/util');
+
+async function getBabelConfig(asset, isSource) {
+  let config = await getBabelRc(asset, isSource);
+  if (!config) {
+    return null;
+  }
+  
+  return {
+    babelVersion: getBabelVersion(config),
+    config
+  };
+}
+
+module.exports = getBabelConfig;
+
+/**
+ * Finds a .babelrc for an asset. By default, .babelrc files inside node_modules are not used.
+ * However, there are some exceptions:
+ *   - if `browserify.transforms` includes "babelify" in package.json (for legacy module compat)
+ *   - the `source` field in package.json is used by the resolver
+ */
+async function getBabelRc(asset, isSource) {
+  // Support legacy browserify packages
+  let pkg = await asset.getPackage();
+  let browserify = pkg && pkg.browserify;
+  if (browserify && Array.isArray(browserify.transform)) {
+    // Look for babelify in the browserify transform list
+    let babelify = browserify.transform.find(
+      t => (Array.isArray(t) ? t[0] : t) === 'babelify'
+    );
+
+    // If specified as an array, override the config with the one specified
+    if (Array.isArray(babelify) && babelify[1]) {
+      return babelify[1];
+    }
+
+    // Otherwise, return the .babelrc if babelify was found
+    return babelify ? await findBabelRc(asset) : null;
+  }
+
+  // If this asset is not in node_modules, always use the .babelrc
+  if (isSource) {
+    return await findBabelRc(asset);
+  }
+
+  // Otherwise, don't load .babelrc for node_modules.
+  // See https://github.com/parcel-bundler/parcel/issues/13.
+  return null;
+}
+
+async function findBabelRc(asset) {
+  // TODO: support babelignore, etc.
+  return await asset.getConfig(['.babelrc', '.babelrc.js'], {
+    packageKey: 'babel'
+  });
+}
+
+function getBabelVersion(babelrc) {
+  if (!babelrc.presets && !babelrc.presets) {
+    return 7;
+  }
+
+  if (
+    (babelrc.presets && hasBabel7Plugin(babelrc.presets)) ||
+    (babelrc.plugins && hasBabel7Plugin(babelrc.plugins))
+  ) {
+    return 7;
+  }
+
+  return 6;
+}
+
+function hasBabel7Plugin(plugins) {
+  return plugins.some(plugin => 
+    getPluginName(plugin).startsWith('@babel/')
```

Comment:
I don't think this way of detecting babel 7 is going to work. If a babelrc only contained third party babel 6 plugins, they wouldn't contain the `@babel/` scope. We need to find a better way of doing this...

Reply:
Not sure what this is for, but a good way of doing this is to check the actual versioning of `babel-core` or `@babel/core`.

Example:
```javascript
const {peerDependencies, dependencies, devDependencies} === require('@babel/preset-react/package.json') ;
let babelVersion;
let babelCore;
if(peerDependencies['@babel/core']) {
  babelCore = peerDependencies['@babel/core'];
}
[babelVersion] = babelCore.split('.');
```

Reply:
Yeah that would be nice, but I doubt most plugins actually declare this dependency in their package.json.

Reply:
For testing it, they would've needed it in their `dependencies` or `devDependencies` (unless it was a rare case where it was a subpackage of a monorepo with the `babel-core` or `@babel/core` version being added to the root). So that could be used as a backup.